### DISCUSSION TITLE
Bigints should work with plot

### DISCUSSION THREAD

**Discussion by llimllib** [observablehq/plot#1099] [Ideas]
I just accidentally threw some bigint numbers at plot, and it failed. It would be great if plot worked with them - why wouldn't it?

In my case, the bigints came via duckdb via parquet; it wasn't something where I was intentionally using them, just the natural output of my ETL process.

The actual failure occurred [here](https://github.com/observablehq/plot/blob/977e0118dad0f05158dbc3c0cab5f92047f449fd/src/scales.js#L455) with `Uncaught (in promise) TypeError: can't convert BigInt to number`.

Is there any reason that it's not feasible for plot to work with bigints?

**Comment by Fil** [comment]
You'd want to replace all coercions from +x to Number(x)?

**Reply by llimllib** [reply]
I've no preference on how coercions work, I'd just want bigints to work with plot

**Comment by josesho** [comment]
I got the same problem via a similar ETL route (loading a parquet file through duckdb on observablehq). So before I handed the data to Plot, I queried the database and passed the data to arquero, then used `Number()` to convert the column for plotting.

```javascript
plotData = aq.from(await db.query(`SELECT xcol, ycol from table`))
             .derive({ycol: aq.escape((d) => Number(ycol))})
```

which Plot then happily accepted. So fairly straightforward to fix on the user's end, but agree with @llimllib if BigInt could be made to work with Plot.

**Reply by Fil** [reply]
Not sure yet how to address this in Plot, but in the meantime:
`SELECT xcol, ycol::INT AS ycol FROM table`
should work too, and not require arquero.
https://observablehq.com/@recifs/duckdb-and-bigint

**Comment by Fil** [comment]
I've opened issue https://github.com/observablehq/plot/issues/1278
